### PR TITLE
Add margin support for Gauge charts

### PR DESCRIPTION
**Pull request by nfranchet**
**What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [ X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
We can't set our own margins

**What is the new behavior?**
I only modify the gauge charts (for my need) to allow margin overriding.

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
I can do the same for other graphs if you think it is useful.

### CODE REVIEW

File: demo/app.component.ts [hunk 1]
```diff
@@ -675,6 +690,11 @@ export class AppComponent implements OnInit {
   gaugeShowAxis: boolean = true;
   gaugeValue: number = 50; // linear gauge value
   gaugePreviousValue: number = 70;
+  gaugeMargin: boolean = false;
```
Comment (marjan-georgiev) [inline demo/app.component.ts]: Let's rename these to be margin, marginLeft, marginRight, marginTop, and marginBottom. We may want to expose the margins on other charts in the future, so they should be generic.

File: demo/app.component.ts [hunk 2]
```diff
@@ -346,6 +346,7 @@ import './demo.scss';
             [showAxis]="gaugeShowAxis"
             [bigSegments]="gaugeLargeSegments"
             [smallSegments]="gaugeSmallSegments"
+            [margin]="gaugeMargin ? [Number(gaugeMarginLeft), Number(gaugeMarginTop), Number(gaugeMarginRight), Number(gaugeMarginBottom)] : null"
```
Comment (marjan-georgiev) [inline demo/app.component.ts]: The order of the margins in the array is [top, right, bottom, left]

File: demo/app.component.ts [hunk 3]
```diff
@@ -605,6 +606,20 @@ import './demo.scss';
...
+
+          <div *ngIf="chart.options.includes('margin')">
+            <label>
+              <input type="checkbox" [checked]="gaugeMargin" (change)="gaugeMargin = $event.target.checked">
+              Show Margin
+            </label> <br />
+          </div>
+
+          <div *ngIf="chart.options.includes('margin') && gaugeMargin">
+            <label>Left</label><input type="text" [(ngModel)]="gaugeMarginLeft"><br />
```
Comment (marjan-georgiev) [inline demo/app.component.ts]: The input type on these should be number

File: demo/app.component.ts [hunk 4]
```diff
@@ -853,4 +873,9 @@ export class AppComponent implements OnInit {
   onLegendLabelClick(entry) {
     console.log('Legend clicked', entry);
   }
+
+  Number(value: any):  number {
```
Comment (marjan-georgiev) [inline demo/app.component.ts]: This will not be needed once input types are changed to `number`

**Comment by nfranchet** [comment]
Done

**Comment by marjan-georgiev** [comment]
Awesome, thank you! ✨

**Pull request merged**